magefiles/magefile.go

```diff
@@ -0,0 +1,47 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	kind "sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cmd"
+)
+
+type Test mg.Namespace
+
+// Runs the unit tests
+func (Test) Unit() error {
+	fmt.Println("running unit tests")
+	return sh.RunV("go", "test", "./...")
+}
+
+// Runs the end-to-end tests in a kind cluster
+func (Test) E2e() error {
+	fmt.Println("running e2e tests")
+	return sh.RunWithV(map[string]string{
+		"PROVISION": "true",
+	}, "go", "run", "github.com/onsi/ginkgo/v2/ginkgo", "--tags=e2e", "--procs=4", "-r", "-v", "--fail-fast", "--randomize-all", "--race", "e2e")
+}
+
+// Cleans up e2e test kube cluster
+func (Test) Clean() error {
```

Comment:
Maybe we should call this E2eClean since it doesn't do anything with unit tests.

Reply:
This should also throw a sane error and not panic if docker is not available